src/gasman.c

```diff
@@ -1295,9 +1308,10 @@ UInt ResizeBag (
     else if (CONST_PTR_BAG(bag) + WORDS_BAG(old_size) == AllocBags) {
         CLEAR_CANARY();
         // check that enough storage for the new bag is available
-        if (EndBags < CONST_PTR_BAG(bag) + WORDS_BAG(new_size)
+        if (SpaceBetweenPointers(EndBags, CONST_PTR_BAG(bag)) < WORDS_BAG(new_size)
               && CollectBags( new_size-old_size, 0 ) == 0 ) {
-            return 0;
+            fputs("gap: cannot extend the workspace any more!!!!!\n",stderr);
```

Comment:
So it seems the number of exclamation marks is supposed to help distinguish these errors?

Reply:
The seems to be the existing pattern, I just carried it on :)